src/ayab/knitter.cpp

```diff
@@ -388,60 +389,39 @@ void Knitter::reqLine(uint8_t lineNumber) {
  */
 bool Knitter::calculatePixelAndSolenoid() {
   uint8_t startOffset = 0;
-  uint8_t laceOffset = 0;
+
+  bool beltShift = BeltShift_t::Shifted == m_beltShift;
 
   switch (m_direction) {
   // calculate the solenoid and pixel to be set
   // implemented according to machine manual
   // magic numbers from machine manual
   case Direction_t::Right:
     startOffset = getStartOffset(Direction_t::Left);
+    m_pixelToSet = m_position - startOffset;
```

Comment:
It looks like this line is identical in both branches of the `switch`, maybe it could be moved below it?

One other potential simplification would be to switch the direction-indexed values in `START_OFFSET` (unless there's a good reason for having `START_OFFSET[Right]` designate the offset for moving to the _left_, but I can't see it) so that you could simply write `startOffset = getStartOffset(m_direction)` and move that out of the `switch` as well.

Reply:
I'll move the m_pixelToSet out but I'm going to leave the start offset stuff for now.